### DISCUSSION TITLE
ImGui Docking giving me only 3 linking errors for 3 particular functions. ImGui Master runs fine.

### DISCUSSION THREAD

**Discussion by Ayan-Banerjee-918** [ocornut/imgui#4532] [New Users Build/Link/Run issues ONLY!]
![Screenshot 2021-09-12 212326](https://user-images.githubusercontent.com/19729019/132996014-a313b218-1b41-47a6-b08a-a0615c802a74.png)
As seen in the screenshot, these are the 3 linking errors that I am getting. Kindly let me know what can be the reason for this and how to fix it.

**Comment by ocornut** [comment]
The only reason i can think of is that you are having a recent version of glfw in your include path but linking an older version (eg the one bundled with dear imgui examples).

**Reply by Ayan-Banerjee-918** [reply]
Thank you @ocornut for pointing it out to me. I downloaded the glfw 3.3.4 version, updated the necessary files, and sure enough its fixed now and working. Thanks again!!

**Reply by BooRar** [reply]
Just fixed that on my side too ... glfw3.lib was stale ...

**Reply by ocornut** [reply]
To clarify, we normally do NOT use those symbols in older GLFW versions:

```cpp
#define GLFW_HAS_WINDOW_ALPHA         (GLFW_VERSION_MAJOR * 1000 + GLFW_VERSION_MINOR * 100 >= 3300) // 3.3+ glfwSetWindowOpacity
#define GLFW_HAS_PER_MONITOR_DPI      (GLFW_VERSION_MAJOR * 1000 + GLFW_VERSION_MINOR * 100 >= 3300) // 3.3+ glfwGetMonitorContentScale
#define GLFW_HAS_MONITOR_WORK_AREA    (GLFW_VERSION_MAJOR * 1000 + GLFW_VERSION_MINOR * 100 >= 3300) // 3.3+ glfwGetMonitorWorkarea
```

```cpp
#if GLFW_HAS_MONITOR_WORK_AREA
        glfwGetMonitorWorkarea(glfw_monitors[n], &x, &y, &w, &h);
...

#if GLFW_HAS_PER_MONITOR_DPI
        glfwGetMonitorContentScale(glfw_monitors[n], &x_scale, &y_scale);
...

#if GLFW_HAS_WINDOW_ALPHA
static void ImGui_ImplGlfw_SetWindowAlpha(ImGuiViewport* viewport, float alpha)
{
    ImGui_ImplGlfw_ViewportData* vd = (ImGui_ImplGlfw_ViewportData*)viewport->PlatformUserData;
    glfwSetWindowOpacity(vd->Window, alpha);
}
#endif
```

**So this can only happen if the HEADERS and linked library are not matching.**
